### DISCUSSION TITLE
Welcome to simple-toshi-ui Discussions!

### DISCUSSION THREAD

**Discussion by chrisbc** [GNS-Science/simple-toshi-ui#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chrisbc** [comment]
Hi @wooneh

**Reply by wooneh** [reply]
@chrisbc

**Comment by wooneh** [comment]
@chrisbc I've compiled a few notes to help with the project set-up:

I usually use Typescript in React projects, since I find that it does a lot for autocompletion and error detection. With a React project, it allows us to pass an interface to components and ensures that those properties are of the correct type (e.g. we would have a RuptureGenerationType and pass it to a card component which will be able to render all the information contained in that type). With JS there would be no error thrown if passing a different type to the component.

I find the error detection a lifesaver for the project I'm working on at the moment, especially with the GraphQL integration that autogenerates all the GraphQL objects into typed objects. 

I would also recommend setting up a linter (ESlint/Prettier) to enforce consistent code practices.

**Reply by chrisbc** [reply]
@wooneh  I need to up my game for dev tools on js/ts!

To date I'm using just Sublime's native js support which does little more than code colouring. I'll take a look for plugins to help out in js/ts words. What editor/ide  do you use??

**Reply by wooneh** [reply]
I use VS Code since I find it super easy to use and customizable. It supports autocompletion and linting with error detection, and runs/compiles code. There are plugins for linting as well which can use the project's linting rules. I also use GitLens which integrates Git, for difftool checking and seeing commit history etc.

**Reply by chrisbc** [reply]
I've been using SublimeText but I hear good things about VSCode. Might give it a try. I'm on Linux/Mac FYI :)

**Comment by wooneh** [comment]
@chrisbc It looks I may have read-only access to the repo, is this something you would be able to check? Thanks!

**Reply by chrisbc** [reply]
oooops - sorry @wooneh I fixed that just now
